api/src/services/roles.ts

```diff
@@ -146,7 +147,45 @@ export class RolesService extends ItemsService {
 		return;
 	}
 
-	override async updateOne(key: PrimaryKey, data: Record<string, any>, opts?: MutationOptions): Promise<PrimaryKey> {
+	private isIpAccessValid(value?: any[] | null): boolean {
+		if (!value || !Array.isArray(value)) return false;
+
+		if (value.length === 0) return true;
+
+		for (const ip of value) {
+			try {
+				getMatch(ip);
```

Comment:
`ip-matching` allows more patterns than we're currently officially listing (wildcards / netmasks) [^1].
I think we should either officially support these patterns or otherwise block them, e.g.
```suggestion
				if (ip.includes('*') || ip.includes('/')) throw new Error();

				getMatch(ip);
```

https://github.com/SchoofsKelvin/ip-matching?tab=readme-ov-file#allowed-patterns

Reply:
I went with a little different approach since we allow CIDR blocks like `10.0.0.0/24` but we now block IP masks like `10.0.1.0/255.0.255.0` (from the `ip-matching` docs) :+1: